httpcache_test.go

```diff
@@ -218,6 +218,27 @@ func TestCacheableMethod(t *testing.T) {
 	}
 }
 
+func TestDontServeHeadResponseToGetRequest(t *testing.T) {
+	resetTest()
+	url := s.server.URL + "/"
+	req, err := http.NewRequest(http.MethodHead, url, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	s.client.Do(req)
```

Comment:
I think you should also check for error here. We wouldn't want this test to pass wrongly if this `Do` fails.

```Go
_, err = s.client.Do(req)
if err != nil {
	t.Fatal(err)
}
```